app/utils/user/index.ts

```diff
@@ -425,6 +425,84 @@ export const convertToAttributesMap = (attributesToConvert: CustomAttributeSet |
     return attributesMap;
 };
 
+export const getDisplayType = (field: CustomProfileFieldModel): string => {
+    if (field.type === 'text' && field.attrs?.value_type !== undefined && field.attrs.value_type !== '') {
+        return field.attrs.value_type;
+    }
+    return field.type;
+};
+
+/**
+ * Get the options from a select or multiselect field as a map of option IDs to option names
+ * @param field - The custom profile field
+ * @returns A record where keys are option IDs and values are option names
+ */
+export const getFieldOptions = (field: CustomProfileFieldModel): Record<string, string> => {
+    if (!field.attrs || (field.type !== 'select' && field.type !== 'multiselect')) {
+        return {};
+    }
+
+    const options = field.attrs.options as Array<{id: string; name: string}> | undefined;
+    if (!options || !Array.isArray(options)) {
+        return {};
+    }
+
+    const optionsMap: Record<string, string> = {};
+    options.forEach((option) => {
+        if (option.id && option.name) {
+            optionsMap[option.id] = option.name;
+        }
+    });
+
+    return optionsMap;
+};
+
+/**
+ * Convert option IDs to display values for select/multiselect fields
+ * @param value - The stored value (option ID or comma-separated IDs)
+ * @param fieldType - The field type ('select' or 'multiselect')
+ * @param optionsMap - Map of option IDs to option names
+ * @returns The display value with option names
+ */
+const convertOptionsToDisplayValue = (value: string, fieldType: string, optionsMap: Record<string, string>): string => {
+    if (!value || !optionsMap) {
+        return value;
+    }
+
+    if (fieldType === 'select') {
+        // Single select: just return the option name for the ID
+        return optionsMap[value] || value;
+    }
+
+    if (fieldType === 'multiselect') {
+        // Multi-select: handle comma-separated IDs or JSON array
+        let optionIds: string[] = [];
+
+        // Try parsing as JSON array first
+        try {
+            const parsed = JSON.parse(value);
```

Comment:
0/5 we have a safeParseJSON, but not sure if it improves anything here.

Reply:
probably not, but I don't mind changing it :)